Android/Samples/libraries/moduletools/src/main/java/com/microsoft/maps/moduletools/ParsingHelpers.java

```diff
@@ -26,4 +27,54 @@ public static void logAltitudeWarning() {
         "Altitude",
         "Unless all positions in a Geometry Object contain an altitude coordinate, all altitudes will be set to 0 at surface level for that Geometry Object.");
   }
+
+  @Nullable
+  public static String verifyPolygonRing(@NonNull ArrayList<Geoposition> positions) {
+    if (positions.size() < 4) {
+      StringBuilder positionsStringBuilder = new StringBuilder();
+      for (int i = 0; i < positions.size() - 1; i++) {
+        Geoposition position = positions.get(i);
+        positionsStringBuilder
+            .append(
+                "["
+                    + position.getLatitude()
+                    + ", "
+                    + position.getLongitude()
+                    + ", "
+                    + position.getAltitude()
+                    + "]")
+            .append(", ");
+      }
+      positionsStringBuilder.append(positions.get(positions.size() - 1));
+      return "Polygon ring must have at least 4 positions, and the first and last position must "
+          + "be the same. Instead saw Geopositions: ["
+          + positionsStringBuilder
+          + "].";
+    }
+
+    Geoposition firstPosition = positions.get(0);
```

Comment:
This is again not clear for me why are you certain that this element exists. Please add check or a comment.

Reply:
For this method, there is a check added that there are at least 4 elements (enough for a valid polygon) in the list positions.

Reply:
Check added that there is at least one element in the list (for printing the Geopositions).